src/typeInference/annotateAST.ts

```diff
@@ -41,11 +41,18 @@ export function annotate(ast: IAST): SequenceType | undefined {
 	switch (ast[0]) {
 		case 'unaryMinusOp':
 			return annotateUnaryMinusOp(ast);
-		case 'addOp':
+		case 'addOp': {
 			const left = annotate(ast[1][1] as IAST);
 			const right = annotate(ast[2][1] as IAST);
 
 			return annotateAddOp(ast, left, right);
+		}
+		case 'subtractOp': {
+			const left = annotate(ast[1][1] as IAST);
+			const right = annotate(ast[2][1] as IAST);
+
+			return annotateSubstractOpp(ast, left, right);
```

Comment:
Maybe change the name to annotateSubtractOp, instead of having a double 'p' at the end. Subtract is also misspelled